src/plugin/explain/explain-plugin.ts

```diff
@@ -0,0 +1,37 @@
+import { QueryResult } from '../../driver/database-connection.js'
+import { RootOperationNode } from '../../query-compiler/query-compiler.js'
+import { ExplainFormat } from '../../util/explainable.js'
+import { AnyRawBuilder, UnknownRow } from '../../util/type-utils.js'
+import {
+  KyselyPlugin,
+  PluginTransformQueryArgs,
+  PluginTransformResultArgs,
+} from '../kysely-plugin.js'
+import { ExplainTransformer } from './explain-transformer.js'
+
+export interface ExplainPluginOptions {
+  enabled?: boolean
+  format?: ExplainFormat
+  raw?: AnyRawBuilder
+}
+
+/**
+ * Plugin that adds `explain` statement before `select`/`insert`/`update`/`replace`/`delete` keywords.
+ */
+export class ExplainPlugin implements KyselyPlugin {
```

Comment:
I don't think this plugin is a good idea. It will lead to wrong output types and will basically break all queries that return data. I don't see this being useful.

Reply:
Yeah, on second thought, no one will ever investigate multiple queries at once. 👍 

r.i.p. ExplainPlugin. 💀